src/main/java/AbstractTask.java

```diff
@@ -0,0 +1,22 @@
+
+public abstract class AbstractTask implements Task {
```

Comment:
I find it interesting that you decided to make a Task interface as well as an abstract class. May I know why you chose such a design?